Makefile

```
@@ -363,25 +363,40 @@ release-build: bundle image-build bundle-build catalog-build ## Build operator,
 .PHONY:
 release-push: image-push bundle-push catalog-push ## Push operator, bundle + catalog images
 
+# It has to be the same namespace as ./config/default/kustomization.yaml -> namespace
+OPERATOR_NAMESPACE ?= backstage-system
+OLM_NAMESPACE ?= olm
+
 .PHONY: deploy-olm
 deploy-olm: ## Deploy the operator with OLM
-	kubectl apply -f config/samples/catalog-operator-group.yaml
-	sed "s/{{VERSION}}/$(subst /,\/,$(VERSION))/g" config/samples/catalog-subscription-template.yaml | sed "s/{{DEFAULT_OLM_NAMESPACE}}/$(subst /,\/,$(DEFAULT_OLM_NAMESPACE))/g" | kubectl apply -f -
+	kubectl -n ${OPERATOR_NAMESPACE} apply -f config/samples/catalog-operator-group.yaml
+	sed "s/{{VERSION}}/$(subst /,\/,$(VERSION))/g" config/samples/catalog-subscription-template.yaml | sed "s/{{OLM_NAMESPACE}}/$(subst /,\/,$(OLM_NAMESPACE))/g" | kubectl -n ${OPERATOR_NAMESPACE} apply -f -
 
 .PHONY: undeploy-olm
 undeploy-olm: ## Un-deploy the operator with OLM
-	-kubectl delete subscriptions.operators.coreos.com backstage-operator
-	-kubectl delete operatorgroup backstage-operator-group
-	-kubectl delete clusterserviceversion backstage-operator.$(VERSION)
+	-kubectl -n ${OPERATOR_NAMESPACE} delete subscriptions.operators.coreos.com backstage-operator
+	-kubectl -n ${OPERATOR_NAMESPACE} delete operatorgroup backstage-operator-group
+	-kubectl -n ${OPERATOR_NAMESPACE} delete clusterserviceversion backstage-operator.v$(VERSION)
```

```suggestion
	-kubectl -n ${OPERATOR_NAMESPACE} delete clusterserviceversion backstage-operator.v$(VERSION)
	-kubectl delete catalogsource backstage-operator -n $(OLM_NAMESPACE)
```

I think we should also delete the catalog source here, to make sure there is nothing all OLM-related resources that were created have been deleted.

---

catalog source is not created by deploy-olm, so it is hardly logical to put it here
we can do something like delete-catalog as complimentary action to update-catalog but I am not sure in it usefulness (+0 actually)

---

Yeah, I agree a separate target to delete the catalog source is not that useful, but I was suggesting this if we agree to make `deploy-olm` depend on `catalog-update`  (see https://github.com/janus-idp/operator/pull/383#discussion_r1636103063), in which case `undeploy-olm` should also delete the catalog source.

---

Like I said, I have no strong opinion on making deploy-olm depend on catalog-update.
If you can tell what is the problem we are solving this way,  we can discuss (better separately from this issue), otherwise - I would use given (Note: we have those two before my PR) and see.